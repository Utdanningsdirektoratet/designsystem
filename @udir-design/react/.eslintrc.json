{
  "extends": ["plugin:@nx/react", "../../.eslintrc.json"],
  "ignorePatterns": ["!**/*", "storybook-static"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "parserOptions": {
        "project": [
          "@udir-design/react/tsconfig.*?.json",
          "@udir-design/react/tsconfig.storybook.json"
        ]
      },
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
