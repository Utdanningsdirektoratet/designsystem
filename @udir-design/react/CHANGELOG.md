## 0.1.0-1 (2024-10-22)

build: don't bundle depenencies


## 0.1.0-0 (2024-10-22)

### ğŸš€ Features

- add Button component ([8d698eb](https://github.com/Utdanningsdirektoratet/designsystem/commit/8d698eb))
