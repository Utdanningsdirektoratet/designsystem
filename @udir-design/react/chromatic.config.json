{
  "onlyChanged": true,
  "projectId": "Project:667e8f07bf467ff4403dfe77",
  "storybookBaseDir": "@udir-design/react",
  "zip": true
}
