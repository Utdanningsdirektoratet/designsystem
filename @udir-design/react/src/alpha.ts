/*
 * This is the entry point for the /alpha sub-module.
 * It exports the alpha-stage components, plus everything in beta (& stable).
 */
export * from './beta';
export * from './components/alpha';
export * from './utilities/hooks/alpha';
