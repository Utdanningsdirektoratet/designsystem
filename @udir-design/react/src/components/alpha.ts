/*
 * This file exports the components that are in the alpha stage of development
 */

export * from './button/Button';
export * from './alert/Alert';
export * from './avatar/Avatar';
export * from './badge/Badge';
export * from './breadcrumbs/Breadcrumbs';
export * from './card/Card';
export * from './checkbox/Checkbox';
export * from './chip/Chip';
export * from './details/Details';
export * from './dialog/Dialog';
export * from './divider/Divider';
export * from './dropdown/Dropdown';
export * from './errorSummary/ErrorSummary';
export * from './field/Field';
export * from './fieldset/Fieldset';
export * from './input/Input';
export * from './link/Link';
export * from './list/List';
export * from './loaders';
export * from './pagination/Pagination';
export * from './popover/Popover';
export * from './radio/Radio';
export * from './search/Search';
export * from './select/Select';
export * from './skipLink/SkipLink';
export * from './switch/Switch';
export * from './table/Table';
export * from './tabs/Tabs';
export * from './tag/Tag';
export * from './textarea/Textarea';
export * from './textfield/Textfield';
export * from './toggleGroup/ToggleGroup';
export * from './tooltip/Tooltip';
export * from './typography';
