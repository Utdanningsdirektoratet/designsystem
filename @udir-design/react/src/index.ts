export * from './button/Button';
export * from './accordion/Accordion';
export * from './alert/Alert';
export * from './avatar/Avatar';
export * from './badge/Badge';
export * from './breadcrumbs/Breadcrumbs';
export * from './card/Card';
export * from './checkbox/Checkbox';
export * from './chip/Chip';
export * from './divider/Divider';
export * from './dropdown/Dropdown';
export * from './error-summary/ErrorSummary';
export * from './fieldset/Fieldset';
export * from './help-text/HelpText';
export * from './link/Link';
export * from './list/List';
export * from './loaders/index';
export * from './modal/Modal';
export * from './pagination/Pagination';
export * from './popover/Popover';
export * from './radio/Radio';
export * from './search/Search';
export * from './select/Select';
export * from './skip-link/SkipLink';
export * from './switch/Switch';
export * from './table/Table';
export * from './tabs/Tabs';
export * from './tag/Tag';
