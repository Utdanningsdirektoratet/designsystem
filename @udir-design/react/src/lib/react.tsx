import styles from './react.module.scss';

export const React: React.FC = () => (
  <div className={styles['container']}>
    <h1>Welcome to React!</h1>
  </div>
);

export default React;
