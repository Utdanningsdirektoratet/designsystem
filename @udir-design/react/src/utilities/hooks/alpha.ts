/*
 * This file exports the hooks that are in the alpha stage of development
 */

export {
  type UsePaginationProps,
  usePagination,
} from './usePagination/usePagination';
export {
  useTableOfContents,
  type useTableOfContentsProps,
} from './useTableOfContents/useTableOfContents';
