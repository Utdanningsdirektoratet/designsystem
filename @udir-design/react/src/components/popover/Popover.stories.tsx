import type { Meta, StoryObj } from '@storybook/react';

import { expect, userEvent, within } from '@storybook/test';
import { TrashIcon } from '@navikt/aksel-icons';
import { Popover, Button, Paragraph } from '../alpha';
import { useEffect, useState } from 'react';

const meta: Meta<typeof Popover> = {
  component: Popover,
  tags: ['alpha'],
  parameters: {
    layout: 'fullscreen',
    customStyles: {
      display: 'flex',
      placeItems: 'end',
      placeContent: 'center',
      padding: '1rem 2rem',
    },
    chromatic: {
      disableSnapshot: false,
    },
  },
  play: async (ctx) => {
    // When not in Docs mode, automatically open the popover
    const canvas = within(ctx.canvasElement);
    const button = canvas.getByRole('button');
    await userEvent.click(button);
    const popover = ctx.canvasElement.querySelector('[popover]');
    await expect(popover).toBeVisible();
  },
};

export default meta;
type Story = StoryObj<typeof Popover>;

export const Preview: Story = {
  args: {
    placement: 'top',
  },
  parameters: {
    customStyles: {
      paddingTop: '5rem',
    },
  },
  render: (args) => (
    <Popover.TriggerContext>
      <Popover.Trigger>My trigger!</Popover.Trigger>
      <Popover {...args}>popover content</Popover>
    </Popover.TriggerContext>
  ),
  play: async ({ canvasElement, step }) => {
    const canvas = within(canvasElement);
    const button = canvas.getByRole('button');
    const dropdown = canvasElement.querySelector('[popover]');

    await step('Open popover', async () => {
      await userEvent.click(button);
      expect(dropdown).toBeVisible();
    });
    await step('Close popover', async () => {
      await userEvent.click(button);
      expect(dropdown).not.toBeVisible();
    });
    await step('Click outside closes dropdown', async () => {
      await userEvent.click(button);
      expect(dropdown).toBeVisible();
      await userEvent.click(document.body);
      expect(dropdown).not.toBeVisible();
    });
  },
};

export const Interactive: Story = {
  parameters: {
    customStyles: {
      padding: '12rem 6rem 1rem',
    },
  },
  render: (args) => {
    return (
      <Popover.TriggerContext>
        <Popover.Trigger data-color="danger" aria-label="Slett rad">
          <TrashIcon title="Slett rad" />
        </Popover.Trigger>
        <Popover {...args} data-color="danger">
          <Paragraph>
            Er du sikker p책 at du vil slette raden? Handlingen kan ikke angres.
          </Paragraph>
          <div
            style={{
              display: 'flex',
              gap: 'var(--ds-size-2)',
              marginTop: 'var(--ds-size-2)',
            }}
          >
            <Button data-size="sm">Ja, slett den</Button>
            <Button data-size="sm" variant="tertiary">
              Avbryt
            </Button>
          </div>
        </Popover>
      </Popover.TriggerContext>
    );
  },
};

export const DottedUnderline: Story = {
  parameters: {
    customStyles: {
      padding: '10rem 6rem 1rem',
    },
  },
  render: (args) => {
    return (
      <Popover.TriggerContext>
        <Paragraph>
          Vi bruker <Popover.Trigger inline>design tokens</Popover.Trigger> for
          책 sikre at vi har konsistent design.
        </Paragraph>
        <Popover {...args} data-color="neutral">
          <Paragraph>
            <strong
              style={{
                display: 'block',
              }}
            >
              Design tokens
            </strong>
            Design tokens er en samling av variabler som definerer designet i et
            designsystem.
          </Paragraph>
        </Popover>
      </Popover.TriggerContext>
    );
  },
};

const VariantsMap: {
  [key: string]: { [key: string]: string };
} = {
  neutralDefault: {
    'data-color': 'neutral',
  },
  neutralTinted: {
    'data-color': 'neutral',
    'data-variant': 'tinted',
  },
  dangerDefault: {
    'data-color': 'danger',
  },
  dangerTinted: {
    'data-color': 'danger',
    'data-variant': 'tinted',
  },
  infoDefault: {
    'data-color': 'info',
  },
  infoTinted: {
    'data-color': 'info',
    'data-variant': 'tinted',
  },
  warningDefault: {
    'data-color': 'warning',
  },
  warningTinted: {
    'data-color': 'warning',
    'data-variant': 'tinted',
  },
};

export const Variants: Story = {
  parameters: {
    customStyles: {
      padding: '5rem 1rem',
    },
  },
  render: function Render(args) {
    const [open, setOpen] = useState(false);

    useEffect(() => setOpen(true), []);

    return (
      <div
        style={{
          display: 'grid',
          gridTemplateColumns: 'repeat(4, 1fr)',
          gap: 'var(--ds-size-2)',
          height: '100%',
          width: '100%',
        }}
      >
        {Object.entries(VariantsMap).map(([key, props], index) => (
          <Popover.TriggerContext key={key}>
            <Popover.Trigger>popover</Popover.Trigger>
            <Popover
              {...args}
              open={open}
              placement={index >= 4 ? 'bottom' : 'top'}
              autoPlacement={false}
              {...props}
            >
              {key}
            </Popover>
          </Popover.TriggerContext>
        ))}
      </div>
    );
  },
  play: () => {
    // Popover already open
  },
};

export const Controlled: Story = {
  parameters: {
    customStyles: {
      padding: '8rem 6rem 1rem',
    },
  },
  render: function Render() {
    const [open, setOpen] = useState(false);

    return (
      <Popover.TriggerContext>
        <Popover.Trigger onClick={() => setOpen(!open)}>
          My trigger
        </Popover.Trigger>
        <Popover
          open={open}
          onClose={() => setOpen(false)}
          data-color="neutral"
        >
          <Paragraph>Er du sikker p책 at du vil slette?</Paragraph>
          <Button
            data-color="danger"
            onClick={() => setOpen(false)}
            data-size="sm"
            style={{ marginTop: 'var(--ds-size-2)' }}
          >
            Slett
          </Button>
        </Popover>
      </Popover.TriggerContext>
    );
  },
};

export const WithoutContext: Story = {
  parameters: {
    customStyles: {
      padding: '8rem 6rem 1rem',
    },
  },
  render: function Render(args) {
    const [open, setOpen] = useState(false);

    return (
      <>
        <Button popovertarget="my-popover" onClick={() => setOpen(!open)}>
          My trigger
        </Button>
        <Popover
          {...args}
          id="my-popover"
          open={open}
          onClose={() => setOpen(false)}
          data-color="neutral"
        >
          <Paragraph>Er du sikker p책 at du vil slette?</Paragraph>
          <Button
            data-color="danger"
            onClick={() => setOpen(false)}
            data-size="sm"
            style={{ marginTop: 'var(--ds-size-2)' }}
          >
            Slett
          </Button>
        </Popover>
      </>
    );
  },
};
