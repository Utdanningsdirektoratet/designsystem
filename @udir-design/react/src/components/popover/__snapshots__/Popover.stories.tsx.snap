// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Color Variants 1`] = `
<div style="display: grid; grid-template-columns: repeat(4, 1fr); gap: 100px; place-items: center; margin-top: var(--ds-size-12);">
  <button
    data-variant="tertiary"
    type="button"
    popovertarget="«rd»"
  >
    popover
  </button>
  <div
    id="«rd»"
    popover="manual"
    data-variant="default"
    data-color="neutral"
    data-placement="top"
    style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
  >
    neutralDefault
  </div>
  <button
    data-variant="tertiary"
    type="button"
    popovertarget="«re»"
  >
    popover
  </button>
  <div
    id="«re»"
    popover="manual"
    data-variant="tinted"
    data-color="neutral"
    data-placement="top"
    style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
  >
    neutralTinted
  </div>
  <button
    data-variant="tertiary"
    type="button"
    popovertarget="«rf»"
  >
    popover
  </button>
  <div
    id="«rf»"
    popover="manual"
    data-variant="default"
    data-color="accent"
    data-placement="top"
    style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
  >
    accentDefault
  </div>
  <button
    data-variant="tertiary"
    type="button"
    popovertarget="«rg»"
  >
    popover
  </button>
  <div
    id="«rg»"
    popover="manual"
    data-variant="tinted"
    data-color="accent"
    data-placement="top"
    style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
  >
    accentTinted
  </div>
  <button
    data-variant="tertiary"
    type="button"
    popovertarget="«rh»"
  >
    popover
  </button>
  <div
    id="«rh»"
    popover="manual"
    data-variant="default"
    data-color="support1"
    data-placement="top"
    style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
  >
    support1Default
  </div>
  <button
    data-variant="tertiary"
    type="button"
    popovertarget="«ri»"
  >
    popover
  </button>
  <div
    id="«ri»"
    popover="manual"
    data-variant="tinted"
    data-color="support1"
    data-placement="top"
    style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
  >
    support1Tinted
  </div>
  <button
    data-variant="tertiary"
    type="button"
    popovertarget="«rj»"
  >
    popover
  </button>
  <div
    id="«rj»"
    popover="manual"
    data-variant="default"
    data-color="support2"
    data-placement="top"
    style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
  >
    support2Default
  </div>
  <button
    data-variant="tertiary"
    type="button"
    popovertarget="«rk»"
  >
    popover
  </button>
  <div
    id="«rk»"
    popover="manual"
    data-variant="tinted"
    data-color="support2"
    data-placement="top"
    style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
  >
    support2Tinted
  </div>
</div>
`;

exports[`Controlled 1`] = `
<button
  data-variant="tertiary"
  type="button"
  popovertarget="«rl»"
>
  Fjern
  <svg
    xmlns="http://www.w3.org/2000/svg"
    width="1em"
    height="1em"
    fill="none"
    viewbox="0 0 24 24"
    focusable="false"
    role="img"
    aria-hidden="true"
  >
    <path
      fill="currentColor"
      d="M18.03 7.03a.75.75 0 0 0-1.06-1.06L12 10.94 7.03 5.97a.75.75 0 0 0-1.06 1.06L10.94 12l-4.97 4.97a.75.75 0 1 0 1.06 1.06L12 13.06l4.97 4.97a.75.75 0 1 0 1.06-1.06L13.06 12z"
    >
    </path>
  </svg>
</button>
<div
  id="«rl»"
  popover="manual"
  data-variant="default"
  data-placement="bottom"
  style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
>
  Er du sikker på at du vil fjerne raden? Handlingen kan ikke angres.
  <div style="display: flex; gap: var(--ds-size-2); margin-top: var(--ds-size-2);">
    <button
      data-variant="primary"
      type="button"
      data-size="sm"
    >
      Ja, fjern
    </button>
    <button
      data-variant="tertiary"
      type="button"
      data-size="sm"
    >
      Avbryt
    </button>
  </div>
</div>
`;

exports[`Dotted Underline 1`] = `
<p data-variant="default">
  I aldersgruppen 1-5 år går 93 prosent av barna i barnehage. I aldersgruppen 3-5 år er andelen 97 prosent. Til sammenligning er
  <button
    popovertarget="«r4»"
    data-popover="inline"
  >
    OECD
  </button>
  -gjennomsnittet for andelen 3-5-åringer i barnehage på 83 prosent (OECD 2021).
</p>
<div
  id="«r4»"
  popover="manual"
  data-variant="default"
  data-color="neutral"
  data-placement="bottom"
  style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
>
  <p data-variant="default">
    <strong>
      Organisasjonen for økonomisk samarbeid og utvikling
    </strong>
    (OCED) er en internasjonal organisasjon som sammarbeider om å fremme økonomisk og sosial vekst og utvikling.
  </p>
</div>
`;

exports[`Interactive 1`] = `
<button
  data-variant="secondary"
  type="button"
  popovertarget="«r2»"
  aria-label="Lukk"
>
  Lukk
  <svg
    xmlns="http://www.w3.org/2000/svg"
    width="1em"
    height="1em"
    fill="none"
    viewbox="0 0 24 24"
    focusable="false"
    role="img"
    aria-hidden="true"
  >
    <path
      fill="currentColor"
      d="M18.03 7.03a.75.75 0 0 0-1.06-1.06L12 10.94 7.03 5.97a.75.75 0 0 0-1.06 1.06L10.94 12l-4.97 4.97a.75.75 0 1 0 1.06 1.06L12 13.06l4.97 4.97a.75.75 0 1 0 1.06-1.06L13.06 12z"
    >
    </path>
  </svg>
</button>
<div
  id="«r2»"
  popover="manual"
  data-variant="default"
  data-placement="bottom"
  style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
>
  <p data-variant="default">
    Vil du lagre endringene dine?
  </p>
  <div style="display: flex; gap: var(--ds-size-2); margin-top: var(--ds-size-2);">
    <button
      data-variant="primary"
      type="button"
      data-size="sm"
    >
      Lagre
    </button>
    <button
      data-variant="tertiary"
      type="button"
      data-size="sm"
    >
      Ikke lagre
    </button>
  </div>
</div>
`;

exports[`Preview 1`] = `
<button
  data-variant="tertiary"
  type="button"
  popovertarget="«r0»"
>
  Fjern
  <svg
    xmlns="http://www.w3.org/2000/svg"
    width="1em"
    height="1em"
    fill="none"
    viewbox="0 0 24 24"
    focusable="false"
    role="img"
    aria-hidden="true"
  >
    <path
      fill="currentColor"
      d="M18.03 7.03a.75.75 0 0 0-1.06-1.06L12 10.94 7.03 5.97a.75.75 0 0 0-1.06 1.06L10.94 12l-4.97 4.97a.75.75 0 1 0 1.06 1.06L12 13.06l4.97 4.97a.75.75 0 1 0 1.06-1.06L13.06 12z"
    >
    </path>
  </svg>
</button>
<div
  id="«r0»"
  popover="manual"
  data-variant="default"
  data-placement="bottom"
  style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
>
  Er du sikker på at du vil fjerne raden? Handlingen kan ikke angres.
  <div style="display: flex; gap: var(--ds-size-2); margin-top: var(--ds-size-2);">
    <button
      data-variant="primary"
      type="button"
      data-size="sm"
    >
      Ja, fjern
    </button>
    <button
      data-variant="tertiary"
      type="button"
      data-size="sm"
    >
      Avbryt
    </button>
  </div>
</div>
`;

exports[`Variants 1`] = `
<div style="display: grid; grid-template-columns: repeat(4, 1fr); gap: 100px; place-items: center; margin-top: var(--ds-size-12);">
  <button
    data-variant="tertiary"
    type="button"
    popovertarget="«r5»"
  >
    popover
  </button>
  <div
    id="«r5»"
    popover="manual"
    data-variant="default"
    data-color="success"
    data-placement="top"
    style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
  >
    successDefault
  </div>
  <button
    data-variant="tertiary"
    type="button"
    popovertarget="«r6»"
  >
    popover
  </button>
  <div
    id="«r6»"
    popover="manual"
    data-variant="tinted"
    data-color="success"
    data-placement="top"
    style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
  >
    successTinted
  </div>
  <button
    data-variant="tertiary"
    type="button"
    popovertarget="«r7»"
  >
    popover
  </button>
  <div
    id="«r7»"
    popover="manual"
    data-variant="default"
    data-color="danger"
    data-placement="top"
    style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
  >
    dangerDefault
  </div>
  <button
    data-variant="tertiary"
    type="button"
    popovertarget="«r8»"
  >
    popover
  </button>
  <div
    id="«r8»"
    popover="manual"
    data-variant="tinted"
    data-color="danger"
    data-placement="top"
    style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
  >
    dangerTinted
  </div>
  <button
    data-variant="tertiary"
    type="button"
    popovertarget="«r9»"
  >
    popover
  </button>
  <div
    id="«r9»"
    popover="manual"
    data-variant="default"
    data-color="info"
    data-placement="top"
    style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
  >
    infoDefault
  </div>
  <button
    data-variant="tertiary"
    type="button"
    popovertarget="«ra»"
  >
    popover
  </button>
  <div
    id="«ra»"
    popover="manual"
    data-variant="tinted"
    data-color="info"
    data-placement="top"
    style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
  >
    infoTinted
  </div>
  <button
    data-variant="tertiary"
    type="button"
    popovertarget="«rb»"
  >
    popover
  </button>
  <div
    id="«rb»"
    popover="manual"
    data-variant="default"
    data-color="warning"
    data-placement="top"
    style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
  >
    warningDefault
  </div>
  <button
    data-variant="tertiary"
    type="button"
    popovertarget="«rc»"
  >
    popover
  </button>
  <div
    id="«rc»"
    popover="manual"
    data-variant="tinted"
    data-color="warning"
    data-placement="top"
    style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
  >
    warningTinted
  </div>
</div>
`;

exports[`Without Context 1`] = `
<button
  data-variant="primary"
  type="button"
  popovertarget="my-popover"
  aria-label="Slett rad"
>
  <svg
    xmlns="http://www.w3.org/2000/svg"
    width="1em"
    height="1em"
    fill="none"
    viewbox="0 0 24 24"
    focusable="false"
    role="img"
    aria-labelledby="title-«rn»"
  >
    <title id="title-«rn»">
      Slett rad
    </title>
    <path
      fill="currentColor"
      fill-rule="evenodd"
      d="M4.5 6.25a.75.75 0 0 0 0 1.5h.805l.876 11.384a1.75 1.75 0 0 0 1.745 1.616h8.148a1.75 1.75 0 0 0 1.745-1.616l.876-11.384h.805a.75.75 0 0 0 0-1.5h-2.75V6A2.75 2.75 0 0 0 14 3.25h-4A2.75 2.75 0 0 0 7.25 6v.25zm5.5-1.5c-.69 0-1.25.56-1.25 1.25v.25h6.5V6c0-.69-.56-1.25-1.25-1.25zm-3.19 3 .867 11.27c.01.13.118.23.249.23h8.148c.13 0 .24-.1.25-.23l.866-11.27zm3.19 2a.75.75 0 0 1 .75.75v6a.75.75 0 0 1-1.5 0v-6a.75.75 0 0 1 .75-.75m4 0a.75.75 0 0 1 .75.75v6a.75.75 0 0 1-1.5 0v-6a.75.75 0 0 1 .75-.75"
      clip-rule="evenodd"
    >
    </path>
  </svg>
</button>
<div
  id="my-popover"
  popover="manual"
  data-variant="default"
  data-color="neutral"
  data-placement="bottom"
  style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
>
  <p data-variant="default">
    Er du sikker på at du vil slette raden? Handlingen kan ikke angres.
  </p>
  <div style="display: flex; gap: var(--ds-size-2); margin-top: var(--ds-size-2);">
    <button
      data-variant="primary"
      type="button"
      data-size="sm"
    >
      Ja, slett den
    </button>
    <button
      data-variant="tertiary"
      type="button"
      data-size="sm"
    >
      Avbryt
    </button>
  </div>
</div>
`;
